src/sema/statements.rs

```diff
@@ -491,10 +493,11 @@ fn statement(
             loops.leave_scope();
 
             res.push(Statement::While(*loc, true, cond, body_stmts));
-
+            context.in_loop = false;
```

Comment:
That's broken.

Reply:
You're right. That should be an integer, not a boolean.